src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -538,17 +538,27 @@ public static decimal Max(decimal val1, decimal val2)
 
         public static double Max(double val1, double val2)
         {
-            if (val1 > val2)
+            // When val1 and val2 are both finite or infinite, return the larger
+            //  * We count +0.0 as larger than -0.0 to match MSVC
+            // When val1 or val2, but not both, are NaN return the opposite
```

Comment:
This is a breaking change, but matches what the C Runtime does and what the IEEE spec defines